include/envoy/stats/stats_options.h

```diff
@@ -44,6 +47,11 @@ class StatsOptions {
    * The max allowed length of a stat suffix.
    */
   virtual size_t maxStatSuffixLength() const PURE;
+
+  /**
+   * The filter regex which limits stat creation.
+   */
+  virtual absl::optional<std::regex> statNameFilter() const PURE;
```

Comment:
not const std::regex& ?

Reply:
I think it makes sense to make this a `const absl::optional<std::regex>&`. Otherwise I'd have to store an `optional<regex>` in the impl but return an `optional<regex&>` which seems odd.